src/main/java/com/checkmarx/flow/service/FlowService.java

```diff
@@ -21,17 +22,26 @@
     @Async("webHook")
     public void initiateAutomation(ScanRequest scanRequest) {
         ScanResults combinedResults = new ScanResults();
+        boolean isAtLeastOneScannerIsEnabled = false;
 
         String effectiveProjectName = projectNameGenerator.determineProjectName(scanRequest);
         scanRequest.setProject(effectiveProjectName);
 
```

Comment:
I would get the list of enabled scanners before entering the scan loop, and then run the loop only on the relevant scanners
If list is empty, log the message and return

Reply:
We implemented it that each scanner has it's own isEnabled() logic, so we do want check it inside the loop.
In case the list is empty we apply a SAST scan by default to maintain the backwards compatibility.

Reply:
List<Scanners> scanners = getListOfScanners(‘here you can run in loop and use the IsEnable implementations, and also put default if empty);

RunScanRequests(scanners);